Library/Homebrew/cmd/info.rb

```diff
@@ -123,8 +123,12 @@ def info_formula(f)
     puts f.desc if f.desc
     puts Formatter.url(f.homepage) if f.homepage
 
-    conflicts = f.conflicts.map(&:name).sort!
-    puts "Conflicts with: #{conflicts*", "}" unless conflicts.empty?
+    conflicts = f.conflicts.map do |c|
+      c.name +
+        (c.reason ? " (because #{c.reason})" : "")
```

Comment:
Make this all a single interpolated string (i.e. `"#{c.name}#{reason}"`) and also do something like `reason = " (because #{c.reason})" if c.reason` to allow the reason to be interpolated.